tests/deprecated_api/test_remove_1-6.py

```diff
@@ -249,3 +249,19 @@ def test_v1_6_0_ddp_plugin_task_idx():
     plugin = DDPPlugin()
     with pytest.deprecated_call(match='Use `DDPPlugin.local_rank` instead'):
         _ = plugin.task_idx
+
+
+def test_v1_6_0_lightning_module_loaded_optimizer_states_dict():
+    from pytorch_lightning.core.lightning import warning_cache
+    model = BoringModel()
+    _ = model.loaded_optimizer_states_dict
+    assert any(
```

Comment:
Could we use `pytest.depreated_call` here instead?

Reply:
I had to do the same for the `LightningModule.datamodule` property deprecation... Not sure why deprecated call doesnt work on these